mindmeld/active_learning/classifiers.py

```diff
@@ -0,0 +1,418 @@
+        return dc_queries_prob_vectors, dc_eval_test
+
+    @staticmethod
+    def _update_eval_stats_domain_level(eval_stats: Dict, dc_eval_test):
+        """Update the eval_stats dictionary with evaluation metrics from the domain
+        classifier.
+
+        Args:
+            eval_stats (Dict): Evaluation metrics to be included in accuracies.json
+            dc_eval_test (): Mindmeld evaluation object for the domain classifier
+        """
+        eval_stats["accuracies"]["overall"] = dc_eval_test.get_stats()["stats_overall"][
+            "accuracy"
+        ]
+        logger.info(
+            "Overall Domain-level Accuracy: %s", eval_stats["accuracies"]["overall"]
+        )
+
+    def intent_classifiers_fit_eval(
+        self,
+        sampled_queries: List[ProcessedQuery],
+        unsampled_queries: List[ProcessedQuery],
+        test_queries: List[ProcessedQuery],
+        domain2id: Dict,
+        intent2id: Dict,
+    ):
+        """Fit and evaluate the domain classifier.
+        Args:
+            sampled_queries (List[ProcessedQuery]): List of Sampled Queries
+            unsampled_queries (List[ProcessedQuery]): List of Unsampled Queries
+            test_queries (List[ProcessedQuery]): List of Test Queries
+            domain2id (Dict): Dictionary mapping domains to IDs
+            intent2id (Dict): Dictionary mapping intents to IDs
+
+        Returns:
+            ic_queries_prob_vectors (List[List]): List of probability distributions
+                for unsampled queries
+            ic_eval_test_dict (Dict): Dictionary mapping a domain (str) to the
+                associated ic_eval_test object
+        """
+        ic_eval_test_dict = {}
+        unsampled_idx_preds_pairs = []
+        for domain in domain2id:
+            # Filter Queries
+            _, filtered_sampled_queries = DataBucket.filter_queries(
+                sampled_queries, domain
+            )
+            (
+                filtered_unsampled_queries_indices,
+                filtered_unsampled_queries,
+            ) = DataBucket.filter_queries(unsampled_queries, domain)
+            _, filtered_test_queries = DataBucket.filter_queries(test_queries, domain)
+            # Train
+            ic = self.nlp.domains[domain].intent_classifier
+            ic.fit(queries=filtered_sampled_queries)
+            # Evaluate Test Queries
+            ic_eval_test = ic.evaluate(queries=filtered_test_queries)
+            if not ic_eval_test:
+                unsampled_idx_preds_pairs.extend(
+                    (i, self._get_zero_prob_vector())
+                    for i in filtered_unsampled_queries_indices
+                )
+                continue
+            ic_eval_test_dict[domain] = ic_eval_test
+            # Get Probability Vectors
+            ic_queries_prob_vectors = MindMeldALClassifier._get_probs(
+                classifier=ic,
+                queries=filtered_unsampled_queries,
+                nlp_component_to_id=intent2id[domain],
```

Comment:
shouldnt this be `intent2id[intent]`?

Reply:
The structure for this mapping is {domain: {intent: id}}. I've renamed this to be domain_to_intent2id to improve the clarity of this variable.